core/frontend/src/tile/map/BingImageryProvider.ts

```diff
@@ -112,7 +112,7 @@ export class BingMapsImageryLayerProvider extends MapLayerImageryProvider {
   }
 
   // construct the Url from the desired Tile
-  public constructUrl(row: number, column: number, zoomLevel: number): string {
+  public async constructUrl(row: number, column: number, zoomLevel: number): Promise<string> {
```

Comment:
Why has this changed to `async`?

Reply:
ArcGIS provider might have to call the token web API to get a valid token, which need to be done in a async manner... because of that I had to make the whole MapLayerImageryProvider hierarchy async.

Reply:
I looked through the implementations of `constructUrl` and couldn't find one that was doing asynchronous stuff. Can you link to one?

Reply:
Here: [https://github.com/imodeljs/imodeljs/blob/5271f2d8860e8a4c4f8c47bc3b5cab027c04aee3/core/frontend/src/tile/map/MapLayerImageryFormats.ts#L658](url)

The call to appendSecurityToken is async.